### PR TITLE
Cased CLI integration

### PR DESCRIPTION
**Pull request by dewski**
This PR wires up `cased-rails` with our Cased CLI approval flow. It includes support for authenticating with Cased, reason being required, session not approved, timed out, or canceled. If the application is configured to record sessions, `rails console` will record playback for non-development or test environments (configurable of course).

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -2,66 +2,67 @@ PATH
   remote: .
   specs:
     cased-rails (0.3.1)
-      cased-ruby (~> 0.3.3)
+      cased-ruby (~> 0.4.0)
```
Comment (dewski) [inline Gemfile.lock:5]: Bump `cased-ruby` to the [new version](https://github.com/cased/cased-ruby/pull/1) we just published.

File: app/views/cased/cli/sessions/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<div id="guard-session-container">
+  <% if guard_session.requested? %>
```
Comment (dewski) [inline app/views/cased/cli/sessions/_form.html.erb:2]: These are all the steps a Guard Session can go through and their respective views for each one.

File: app/views/cased/cli/sessions/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  <% elsif guard_session.denied? %>
+    <%= render 'cased/cli/sessions/steps/denied', guard_session: guard_session %>
+  <% elsif guard_session.canceled? %>
+    <%= render 'cased/cli/sessions/steps/canceled', guard_session: guard_session %>
+  <% elsif guard_session.timed_out? %>
+    <%= render 'cased/cli/sessions/steps/timed_out', guard_session: guard_session %>
+  <% elsif guard_session.reason_required? %>
+    <%= render 'cased/cli/sessions/steps/reason_required', guard_session: guard_session %>
+  <% else %>
+    <%= render 'cased/cli/sessions/steps/create', guard_session: guard_session %>
```
Comment (dewski) [inline app/views/cased/cli/sessions/_form.html.erb:13]: The create step is special in that we need to preserve any form parameters even when unauthenticated, we submit the hidden form to render either the `requested`, `reason_required`, or `denied`/`canceled`/`timed_out` state.

File: app/views/cased/cli/sessions/_guard_session.json.jbuilder [hunk 4]
```diff
@@ -0,0 +1,27 @@
+json.id guard_session.id
```
Comment (dewski) [inline app/views/cased/cli/sessions/_guard_session.json.jbuilder:1]: This view is used when polling for the approved/denied/timed out/canceled state.

File: app/views/cased/cli/sessions/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      <div class="text-center py-4 flex justify-center space-x-2">
+        <div>
+          Logged in as <span id="cased-user"><%= cased_authorization %></span>.
+        </div>
+        <%= button_to 'Logout', cased.logout_path, form: { id: 'cased-logout' }, method: :delete, remote: true, class: 'p-0 border-none bg-transparent text-blue-500 shadow-none cursor-pointer' %>
+      </div>
+    </div>
+    <div id="cased-logged-out" class="<%= 'hidden' if cased_authorization? %>">
+      <div class="bg-white py-8 px-4 shadow sm:rounded-lg sm:px-10">
+        <%= link_to 'Sign in to Cased', "#{Cased.config.url}/login/connect/#{Cased.config.guard_application_key}?return_to=#{cased.authorizations_url}", id: 'cased-authenticate', class: 'w-full flex justify-center py-2 px-4 border border-transparent rounded-md shadow-sm text-sm font-medium text-white bg-gray-600 hover:bg-gray-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-gray-500' %>
```
Comment (dewski) [inline app/views/cased/cli/sessions/new.html.erb:20]: This URL ends up looking like:

```
https://app.cased.com/login/connect/guard_application_1234?return_to=https://admin.application.com/cased/authorizations/callback
```

**Review by rikasun — APPROVED** [review]
🙌🏼   🥳

**Pull request merged**